### PR TITLE
=doc #18706 re-enable RangeDirectivesExampleSpec

### PR DESCRIPTION
**Pull request by johanandren**
RangeDirectivesExamplesSpec uncommented and updated to work

Refs #18706

### CODE REVIEW

File: akka-docs-dev/rst/scala/code/docs/http/scaladsl/server/directives/RangeDirectivesExamplesSpec.scala [hunk 1]
```diff
@@ -1,38 +1,56 @@
...
+import akka.util.ByteString
+
+import headers._
+import scala.concurrent.Await
+import scala.concurrent.duration._
+
+class RangeDirectivesExamplesSpec extends RoutingSpec {
+
+  override def testConfig: Config =
+    ConfigFactory.parseString("akka.http.routing.range-coalescing-threshold=2").withFallback(super.testConfig)
```
Comment (ktoso) [inline akka-docs-dev/rst/scala/code/docs/http/scaladsl/server/directives/RangeDirectivesExamplesSpec.scala]: Can you add an explanation about this to `range-directives/index.rst` and make it a section maybe, and then `:ref:` to it from `withRangeSupport.rst`? Thanks!

**Comment by ktoso** [comment]
Awesome, thanks @johanandren!
Let's add some docs and then merge :)

**Comment by johanandren** [comment]
Turns out there already was docs for the configuration of the ```range-coalescing-threshold``` setting, I had just missed it.

**Comment by johanandren** [comment]
And to be more complete, it is documented in ```withRangeSupport.rst```

**Comment by ktoso** [comment]
Hah, that's what I get by drowning in directives for a week :wink: 
Let's update it a bit however, and make the mentioned setting be explicit about the entire setting, i.e. `akka.http.routing.range-coalescing-threshold` instead of `range-coalescing-threshold` (I myself confuse where a setting is sometimes, so let's make it simple for people)

**Comment by ktoso** [comment]
@johanandren in `withRngeSupport.rst` there's `TODO (../../../../code/docs/http/scaladsl/server/directives/RangeDirectivesExamplesSpec.scala)` we need to fix that to be 

```
.. includecode2:: ../../../../code/docs/http/scaladsl/server/directives/RangeDirectivesExamplesSpec.scala
   :snippet: ????? (test name)
```

File: akka-docs-dev/rst/scala/code/docs/http/scaladsl/server/directives/RangeDirectivesExamplesSpec.scala [hunk 2]
```diff
@@ -1,38 +1,56 @@
...
+
+  "withRangeSupport" in {
+    val route =
```
Comment (ktoso) [inline akka-docs-dev/rst/scala/code/docs/http/scaladsl/server/directives/RangeDirectivesExamplesSpec.scala]: please add `// akka.http.routing.range-coalescing-threshold = 2` here, so people understand "oh, that's why it's 2"

File: akka-docs-dev/rst/scala/http/routing-dsl/directives/range-directives/withRangeSupport.rst [hunk 3]
```diff
@@ -42,4 +42,4 @@ See also: https://tools.ietf.org/html/rfc7233
 Example
 -------
 
-TODO (../../../../code/docs/http/scaladsl/server/directives/RangeDirectivesExamplesSpec.scala)
+.. includecode2:: ../../../../code/docs/http/scaladsl/server/directives/RangeDirectivesExamplesSpec.scala
```
Comment (ktoso) [inline akka-docs-dev/rst/scala/http/routing-dsl/directives/range-directives/withRangeSupport.rst]: Almost... :)
Needs:

```
.. includecode2:: ../../../../code/docs/http/scaladsl/server/directives/RangeDirectivesExamplesSpec.scala
   :snippet: withRangeSupport
```

Comment (johanandren) [inline akka-docs-dev/rst/scala/http/routing-dsl/directives/range-directives/withRangeSupport.rst]: dang!

File: akka-docs-dev/rst/scala/code/docs/http/scaladsl/server/directives/RangeDirectivesExamplesSpec.scala [hunk 4]
```diff
@@ -1,38 +1,59 @@
...
+      }
+
+    Get() ~> addHeader(Range(ByteRange(3, 4))) ~> route ~> check {
+      headers should contain(`Content-Range`(ContentRange(3, 4, 8)))
+      status shouldEqual StatusCodes.PartialContent
+      responseAs[String] shouldEqual "DE"
+    }
+
+
+    // we set "akka.http.routing.range-coalescing-threshold = 2"
```
Comment (ktoso) [inline akka-docs-dev/rst/scala/code/docs/http/scaladsl/server/directives/RangeDirectivesExamplesSpec.scala]: great, thanks

File: akka-docs-dev/rst/scala/code/docs/http/scaladsl/server/directives/RangeDirectivesExamplesSpec.scala [hunk 5]
```diff
@@ -1,38 +1,59 @@
...
+    // above to make sure we get two BodyParts
```
Comment (ktoso) [inline akka-docs-dev/rst/scala/code/docs/http/scaladsl/server/directives/RangeDirectivesExamplesSpec.scala]: isn't it "to make sure we get parts of size 2"?

**Comment by ktoso** [comment]
LGTM after https://github.com/akka/akka/pull/18739/files#r42472103 is cleared up

File: akka-docs-dev/rst/scala/code/docs/http/scaladsl/server/directives/RangeDirectivesExamplesSpec.scala (hunk 5 above)
Comment (johanandren) [inline akka-docs-dev/rst/scala/code/docs/http/scaladsl/server/directives/RangeDirectivesExamplesSpec.scala]: No, the threshold is to not merge ranges where there are more than 'n' in-between, in this case the first two ranges 0-1 and 1-2 gets merged but then there is a gap larger than 2 so the 6-7 becomes a separate body part.

Comment (ktoso) [inline akka-docs-dev/rst/scala/code/docs/http/scaladsl/server/directives/RangeDirectivesExamplesSpec.scala]: Ah, now I finally understand it, thanks!

**Comment by ktoso** [comment]
LGTM, thanks

**Pull request merged**